samples/tfm_integration/psa_crypto/src/psa_crypto.c

```diff
@@ -91,105 +107,421 @@ static psa_status_t crp_get_pub_key(psa_key_id_t key_id,
+    psa_set_key_id(&key_attributes, key_id);
+    psa_set_key_usage_flags(&key_attributes, key_usage);
+    psa_set_key_lifetime(&key_attributes, PSA_KEY_LIFETIME_PERSISTENT);
+    psa_set_key_algorithm(&key_attributes, alg);
+    psa_set_key_type(&key_attributes, key_type);
+
+    /* Import the private key, creating the persistent key on success */
+    status = al_psa_status(
+        psa_import_key(&key_attributes, key_data, key_len, &key_handle),
+        __func__);
+    if (status != PSA_SUCCESS) {
+        LOG_ERR("Failed to import key.");
+        goto err;
+    }
+
+    /* Close the key to free up the volatile slot. */
+    status = al_psa_status(
+        psa_close_key(key_handle),
+        __func__);
+    if (status != PSA_SUCCESS) {
+        LOG_ERR("Failed to close persistent key.");
+        goto err;
+    }
+
+    /* Try to retrieve the public key. */
+    status = crp_get_pub_key(key_id, data_out, sizeof(data_out), &data_len);
+
+    /* Export the private key if usage includes PSA_KEY_USAGE_EXPORT. */
+    if (key_usage & PSA_KEY_USAGE_EXPORT) {
+        /* Re-open the persisted key based on the key ID. */
+        status = al_psa_status(
+            psa_open_key(key_id, &key_handle),
+            __func__);
+        if (status != PSA_SUCCESS) {
+            LOG_ERR("Failed to open persistent key #%d", key_id);
+            goto err;
+        }
+
+        /* Read the original (private) key data back. */
+        status = al_psa_status(
+            psa_export_key(key_handle, data_out,
+                       sizeof(data_out), &data_len),
+            __func__);
+        if (status != PSA_SUCCESS) {
+            LOG_ERR("Failed to export key.");
+            goto err;
+        }
+
+        /* Check key len. */
+        if (data_len != key_len) {
+            LOG_ERR("Unexpected number of bytes in exported key.");
+            goto err;
+        }
+
+        /* Verify that the exported private key matches input data. */
+        comp_result = memcmp(data_out, key_data, key_len);
+        if (comp_result != 0) {
+            LOG_ERR("Imported/exported private key mismatch.");
+            goto err;
+        }
+
+        /* Display the private key. */
+        LOG_INF("Private key data:");
+		al_dump_log();
+        sf_hex_tabulate_16(&crp_fmt, data_out, data_len);
+
+        /* Close the key to free up the volatile slot. */
+        status = al_psa_status(
+            psa_close_key(key_handle),
+            __func__);
+        if (status != PSA_SUCCESS) {
+            LOG_ERR("Failed to close persistent key.");
+            goto err;
+        }
+    }
+
+    return status;
+err:
 	al_dump_log();
+	return status;
+}
 
-	/* Setup the key's attributes before the creation request. */
-	psa_set_key_id(&key_attributes, key_id);
-	psa_set_key_usage_flags(&key_attributes, key_usage);
-	psa_set_key_algorithm(&key_attributes, alg);
-	psa_set_key_type(&key_attributes, key_type);
+#else // CONFIG_PSA_IMPORT_KEY
```

Comment:
Lines 211 and 316 have the same macro (`// CONFIG_PSA_IMPORT_KEY`), so it isn't clear which code block hands which case. Clarify which with a distinction between the two cases.

Reply:
I thought adding the condition next to the `#else` would make it easier to understand. I've added a `!` before the condition in comment. Is that helpful?

The docstrings ([if](https://github.com/zephyrproject-rtos/zephyr/pull/40103/files/9393d7431d5928fc7d14eb5cc11e73408e943ab2#diff-eb71373f32415464bd1de047b52fafb9ae166ccf3e334e75ddd819c7bc1b7e9dR97) and [else](https://github.com/zephyrproject-rtos/zephyr/pull/40103/files/9393d7431d5928fc7d14eb5cc11e73408e943ab2#diff-eb71373f32415464bd1de047b52fafb9ae166ccf3e334e75ddd819c7bc1b7e9dR212)) for the functions define the functionality of the code block.